graphql-server/pkg/dataloaders/proposal.go

```diff
@@ -0,0 +1,106 @@
+package dataloaders
+
+import (
+	"time"
+
+	godataloader "github.com/cychiuae/go-dataloader"
+	"github.com/oursky/likedao/pkg/models"
+	"github.com/oursky/likedao/pkg/queries"
+)
+
+type ProposalDataloader interface {
+	LoadProposalConnection(input models.QueryProposalsInput) (*models.ProposalConnection, error)
+	LoadProposalConnectionByStatus(input StatusProposalConnectionLoaderInput) (*models.ProposalConnection, error)
+	LoadProposalConnectionByRelatedAddress(input RelatedAddressProposalConnectionLoaderInput) (*models.ProposalConnection, error)
+}
+
+type ProposalConnectionDataloader interface {
+	Load(input models.QueryProposalsInput) (*models.ProposalConnection, error)
+}
+
+type StatusProposalConnectionLoaderInput struct {
+	Status     models.ProposalStatus
+	QueryInput models.QueryProposalsInput
+}
+type StatusProposalConnectionDataloader interface {
+	Load(input StatusProposalConnectionLoaderInput) (*models.ProposalConnection, error)
+}
+
+type RelatedAddressProposalConnectionLoaderInput struct {
+	RelatedAddress string
+	QueryInput     models.QueryProposalsInput
+}
+type RelatedAddressProposalConnectionDataloader interface {
+	Load(input RelatedAddressProposalConnectionLoaderInput) (*models.ProposalConnection, error)
+}
+
+type IProposalDataloader struct {
+	proposalConnectionLoader               ProposalConnectionDataloader
+	statusProposalConnectionLoader         StatusProposalConnectionDataloader
+	relatedAddressProposalConnectionLoader RelatedAddressProposalConnectionDataloader
+}
+
+func NewProposalDataloader(proposalQuery queries.IProposalQuery) ProposalDataloader {
```

Comment:
@hochiw let's stick to db models at data loader level; better map to connection type at resolver?

Reply:
don't quite get - since it's 3 different queries we need 3 loaders? and we pick the suitable one depending on the input at resolver

Reply:
actually we don't need to use dataloader for this query as there is no n+1 issue? we only need to implement dataloader for resolving proposal(s) by id(s)

Reply:
@chihimng you right, updated 🙏